js/src/common/extend.ts

```diff
@@ -1,3 +1,10 @@
+export type KeyOfType<T extends object, Type> = Exclude<
+  {
+    [K in keyof T]: T[K] extends Type ? K : never;
+  }[keyof T],
+  undefined
```

Comment:
Why `undefined` and not `never`? Aren't we already filtering for nonexistent keys through the type map?

Reply:
I think this is the old equivalent of the `-?:` in my type.

Reply:
i have no idea how that code works